src/lib/forward.ts

```diff
@@ -5,24 +5,45 @@ import {
   ZWaveNodeEvents,
   ZWaveNodeMetadataUpdatedArgs,
 } from "zwave-js";
-import { CommandClasses, ConfigurationMetadata } from "@zwave-js/core";
+import {
+  CommandClasses,
+  ConfigurationMetadata,
+  ZWaveLogInfo,
+} from "@zwave-js/core";
 import { OutgoingEvent } from "./outgoing_message";
 import { dumpConfigurationMetadata, dumpMetadata, dumpNode } from "./state";
-import { Client, ClientsController } from "./server";
+import { Client, ClientsController, ZwavejsServer } from "./server";
 
 export class EventForwarder {
   /**
    * Only load this once the driver is ready.
    *
    * @param clients
    */
-  constructor(public clients: ClientsController) {}
+  constructor(
+    public clients: ClientsController,
+    private server: ZwavejsServer
+  ) {}
 
   start() {
     this.clients.driver.controller.nodes.forEach((node) =>
       this.setupNode(node)
     );
 
+    // Forward logging events on to clients that are currently
+    // receiving logs
+    this.server.on("logging", (message: string) => {
```

Comment:
We should attach this only when people are listening, and remove it when last listener is gone. That way we don't create a bunch of overhead.

Reply:
Not sure yet how to instrument this on the listener side but will think it through. Is the overhead on the listener side or on the emitter side? Or is the concern that both are always up with this approach?